File: app/src/main/java/br/com/alura/estoque/retrofit/service/ProdutoService.java
package br.com.alura.estoque.retrofit.service;

import java.util.List;

import br.com.alura.estoque.model.Produto;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface ProdutoService {

    @GET("produto")
    Call<List<Produto>> buscaTodos();

    @POST("produto")
    Call<Produto> salva(@Body Produto produto);

    @PUT("produto/{id}")
    Call<Produto> edita(@Path("id") long id,
                        @Body Produto produto);

}

File: src/main/kotlin/br/com/alura/estoqueapi/service/ProdutoService.kt
package br.com.alura.estoqueapi.service

import br.com.alura.estoqueapi.model.Produto
import br.com.alura.estoqueapi.repository.ProdutoRepository
import org.springframework.stereotype.Service

@Service
class ProdutoService(
        val produtoRepository: ProdutoRepository) {

    fun buscaTodos(): List<Produto> {
        return produtoRepository.findAll().toList()
    }

    fun salva(id: Long = 0, produto: Produto): Produto {
        val produtoComIdAtualizado = produto.copy(id = id)
        return produtoRepository.save(produtoComIdAtualizado)
    }

    fun temProduto(id: Long): Boolean {
        return produtoRepository.existsById(id)
    }

    fun remove(id: Long) {
        produtoRepository.deleteById(id)
    }

}